File: api/src/Repository/RoleRepository.php
<?php
declare(strict_types=1);

namespace App\Repository;

use Doctrine\ORM;

use Doctrine\ORM\EntityRepository as EntityRepository;
use Doctrine\ORM\EntityManager;
use App\Entity\Role;

class RoleRepository
{
    private EntityManager $em;
    private $repo;

    function __construct(EntityManager $entityManager) {
        $this->em = $entityManager;
        $this->repo = $entityManager->getRepository(Role::class);
    }

    function getActiveRoles(): array {
        $query =$this->em->createQuery("SELECT t FROM App\Entity\Role t WHERE t.active = 1");

        return $query->getResult();
    }

    function getById(int $id): ?Role {
        $query =$this->em->createQuery("SELECT t FROM App\Entity\Role t WHERE t.active = 1 and t.id = :id");
        $query->setParameter('id', $id);

        return $query->getOneOrNullResult();
    }

    function getByName(string $name): ?Role {
        $query =$this->em->createQuery("SELECT t FROM App\Entity\Role t WHERE t.active = 1 and t.name = :name");
        $query->setParameter('name', $name);

        return $query->getOneOrNullResult();
    }

}

File: api/src/main/java/be/jimmyd/cm/repositories/RoleRepository.java
package be.jimmyd.cm.repositories;

import be.jimmyd.cm.entities.Role;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface RoleRepository extends JpaRepository<Role, Long> {

    @Query("FROM Role WHERE active = true")
    List<Role> getActiveRoles();

    @Query("FROM Role WHERE name = :name")
    Role getByName(@Param("name") String name);
}
